src/Native/Runtime/arm64/AllocFast.asm

```diff
@@ -81,7 +81,7 @@ RhpNewFast_RarePath
 
         ;; If the object is bigger than RH_LARGE_OBJECT_SIZE, we must publish it to the BGC
         ldr         w1, [x19, #OFFSETOF__EEType__m_uBaseSize]
-        movk        x2, #(RH_LARGE_OBJECT_SIZE & 0xFFFF)
+        movz        x2, #(RH_LARGE_OBJECT_SIZE & 0xFFFF)
```

Comment:
Cc @AntonLapounov who tends to troubleshoot weird ARM bugs on .NET Native - we have this bug in ProjectNRel.

Reply:
This is just a information for someone else. Nothing to do for me?

Reply:
Yep - this code originates from [.NET Native](https://docs.microsoft.com/en-us/dotnet/framework/net-native/getting-started-with-net-native) but there's no backflow of fixes from this repo to .NET Native so if we want to fix anything there, it needs to be done separately. It's good to be at least aware that an issue exists.

Reply:
Nice catch. I guess in .NET Native `RhpGcAlloc` does not modify `x2`, which is why we have never hit this issue.